sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,210 @@
+import com.azure.cosmos.util.Beta;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * Group configuration which will be used in Throughput control.
+ */
+@Beta(value = Beta.SinceVersion.V4_10_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+
+    private final static boolean DEFAULT_USE_BY_DEFAULT = false;
+    private final static ThroughputControlMode DEFAULT_CONTROL_MODE = ThroughputControlMode.LOCAL;
+
+    private ThroughputControlMode controlMode;
+    private String groupName;
+    private CosmosAsyncContainer targetContainer;
+    private Integer targetThroughput;
+    private Double targetThroughputThreshold;
+    private boolean useByDefault;
+
+    public ThroughputControlGroup() {
+        this.controlMode = DEFAULT_CONTROL_MODE;
+        this.useByDefault = DEFAULT_USE_BY_DEFAULT;
+    }
+
+    /**
+     * Get throughput group control mode.
+     *
+     * By default, it will be local control mode.
+     *
+     * @return the {@link ThroughputControlMode}.
+     */
+    public ThroughputControlMode getControlMode() {
+        return this.controlMode;
+    }
+
+    /**
+     * Set throughput group control mode to local.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup localControlMode() {
+        this.controlMode = ThroughputControlMode.LOCAL;
+        return this;
+    }
+
+    /**
+     * Set the throughput control group name.
+     *
+     * @param groupName
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup groupName(String groupName) {
+        checkArgument(StringUtils.isNotEmpty(groupName), "Group name can not be null or empty");
+        this.groupName = groupName;
+
+        return this;
+    }
+
+    /**
+     * Get the throughput control group name.
+     *
+     * @return the group name.
+     */
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    /**
+     * Get the throughput control group target container.
+     *
+     * @return the {@link CosmosAsyncContainer}.
+     */
+    public CosmosAsyncContainer getTargetContainer() {
+        return this.targetContainer;
+    }
+
+    /**
+     * Set the throughput control group target container.
+     *
+     * @param targetContainer the target container for the throughput control group.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup targetContainer(CosmosAsyncContainer targetContainer) {
+        checkArgument(targetContainer != null, "Target container cannot be null");
+        this.targetContainer = targetContainer;
+        return this;
+    }
+
+    /**
+     * Get throughput control group target throughput.
+     *
+     * @return the target throughput.
+     */
+    public Integer getTargetThroughput() {
```

Comment:
yup, since we allow the customer to either use targetThroughput or targetThroughputThreshold, so I think we should allow null value

Reply:
please add in the documentation that it may return null value.

Reply:
Added the following comments:
/**
     * Get throughput control group target throughput.
     * Since we allow either TargetThroughput or TargetThroughputThreshold, this value can be null.
     *
     * By default, it is null.
     *
     * @return the target throughput.
     */